pkg/render/logstorage/esgateway/esgateway.go

```diff
@@ -0,0 +1,427 @@
+package esgateway
+
+import (
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/dns"
+	"github.com/tigera/operator/pkg/render"
+	relasticsearch "github.com/tigera/operator/pkg/render/common/elasticsearch"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+)
+
+const (
+	EsGatewayName                    = "tigera-es-gateway"
+	EsGatewayServiceAccountName      = "tigera-es-gateway"
+	EsGatewayRole                    = "tigera-es-gateway"
+	EsGatewaySecretKeyName           = "key"
+	EsGatewaySecretCertName          = "tls.crt"
+	EsGatewayVolumeName              = "tigera-es-gateway-certs"
+	EsGatewayElasticServiceName      = "tigera-secure-es-http"
+	EsGatewayKibanaServiceName       = "tigera-secure-kb-http"
+	EsGatewayKibanaPublicCertSecret  = "tigera-secure-kb-http-certs-public"
+	EsGatewayElasticPublicCertSecret = "tigera-secure-es-http-certs-public"
+	EsGatewayElasticUserSecret       = "tigera-es-gateway-elasticsearch-access"
+	EsGatewayTLSSecret               = "tigera-es-gateway-tls"
+	EsGatewayPortName                = "es-gateway-port"
+	EsGatewayPort                    = 5554
+
+	ElasticsearchHTTPSEndpoint = "https://tigera-secure-internal-es-http.tigera-elasticsearch.svc:9200"
+	ElasticsearchPort          = 9200
+	KibanaHTTPSEndpoint        = "https://tigera-secure-internal-kb-http.tigera-kibana.svc:5601"
+	KibanaPort                 = 5601
+
+	tlsSecretHashAnnotation = "hash.operator.tigera.io/tls-secret"
+)
+
+func EsGateway(
+	logStorage *operatorv1.LogStorage,
+	installation *operatorv1.InstallationSpec,
+	pullSecrets []*corev1.Secret,
+	esAdminUserSecret *corev1.Secret,
+	kibanaPublicCertSecret *corev1.Secret,
+	esGatewayTLSSecret *corev1.Secret,
+	kubeControllersSecret *corev1.Secret,
+	clusterDomain string,
+) render.Component {
+	var esGatewayTLSSecrets []*corev1.Secret
+	esGatewayTLSAnnotations := map[string]string{}
+
+	// tigera-es-gateway-elasticsearch-access
+	if esAdminUserSecret != nil {
+		esGatewayElasticUserSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      EsGatewayElasticUserSecret,
+				Namespace: render.ElasticsearchNamespace,
+			},
+			Data: map[string][]byte{
+				"username": []byte("elastic"),
+				"password": esAdminUserSecret.Data["elastic"],
+			},
+		}
+		esGatewayTLSSecrets = append(esGatewayTLSSecrets, esGatewayElasticUserSecret)
+	}
+
+	// tigera-es-gateway-tls
+	esGatewayTLSSecrets = append(esGatewayTLSSecrets, esGatewayTLSSecret)
```

Comment:
We need to make sure that we don't write back resources that can be create outside of the render function. This TLS secret may be created by a customer, or something other than the operator, so we shouldn't write back to it (it may change the owner reference).

You can still copy these secrets to other namespace though.